### PR TITLE
Update loops to prevent crash on some GLTF models.

### PR DESCRIPTION
**Pull request by ksuprynowicz**
Fixes https://github.com/vircadia/vircadia/issues/1542

### CODE REVIEW

**Review by HifiExperiments — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-10.15, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-Interface-PR1545-ab98c02-_9a3a22d3.dmg

**build (macOS-10.15, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-ab98c02-_9a3a22d3.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-ab98c02-_9a3a22d3.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-ab98c02-.tar_9a3a22d3.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-ab98c02-.tar_9a3a22d3.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-ab98c02-_9a3a22d3.exe

**build (self-hosted_debian-11_aarch64, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-ab98c02-_9a3a22d3.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-ab98c02-.tar_9a3a22d3.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-ab98c02-.tar_9a3a22d3.Z

File: libraries/model-serializers/src/GLTFSerializer.cpp [hunk 1]
```diff
@@ -1356,17 +1356,17 @@ bool GLTFSerializer::buildGeometry(HFMModel& hfmModel, const hifi::VariantHash&
 
                 part.triangleIndices.append(validatedIndices);
 
-                for (int n = 0; n < vertices.size(); n = n + verticesStride) {
+                for (int n = 0; n + 2 < vertices.size(); n = n + verticesStride) {
```
Comment (ctrlaltdavid) [inline libraries/model-serializers/src/GLTFSerializer.cpp:1359]:
```suggestion
                for (int n = 0; n + verticesStride - 1 < vertices.size(); n = n + verticesStride) {
```

File: libraries/model-serializers/src/GLTFSerializer.cpp [hunk 2]
```diff
@@ -1356,17 +1356,17 @@ bool GLTFSerializer::buildGeometry(HFMModel& hfmModel, const hifi::VariantHash&
...
                     mesh.vertices.push_back(glm::vec3(vertices[n], vertices[n + 1], vertices[n + 2]));
                 }
 
-                for (int n = 0; n < normals.size(); n = n + normalStride) {
+                for (int n = 0; n + 2 < normals.size(); n = n + normalStride) {
```
Comment (ctrlaltdavid) [inline libraries/model-serializers/src/GLTFSerializer.cpp:1363]:
```suggestion
                for (int n = 0; n + normalStride - 1 < normals.size(); n = n + normalStride) {
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-10.15, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-c946cf2-_a7883fc5.dmg

**build (macOS-10.15, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-Interface-PR1545-c946cf2-_a7883fc5.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-c946cf2-.tar_a7883fc5.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-c946cf2-.tar_a7883fc5.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-c946cf2-_a7883fc5.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-c946cf2-_a7883fc5.exe

**build (self-hosted_debian-11_aarch64, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-c946cf2-.tar_a7883fc5.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-c946cf2-_a7883fc5.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1545/Vircadia-PR1545-c946cf2-.tar_a7883fc5.Z

**Comment by Aitolda** [comment]
Do you have any of the models that were causing the crash for testing?

**Comment by ksuprynowicz** [comment]
> Do you have any of the models that were causing the crash for testing?
Here's the link to them:
https://drive.google.com/file/d/1fkCb173ts0QlMuZjtMTdJpr0rHIKqQ51/view?usp=sharing
One of them is immediately crashing current release.

**Comment by ksuprynowicz** [comment]
I tested it on Linux, and it fixed the bug. This PR works fine and on the same world current master branch crashes immediately.

**Comment by Aitolda** [comment]
> Here's the link to them:
https://drive.google.com/file/d/1fkCb173ts0QlMuZjtMTdJpr0rHIKqQ51/view?usp=sharing
One of them is immediately crashing current release

**Comment by Aitolda** [comment]
I could not reproduce the crash on the current version using Windows 10 with any of the models  with either the current version (Selene) OR this PR. However, I went to various domains and had no issues with models that weren't already present with Selene, so it doesn't appear to have broken anything either.  Perhaps this bug is unique to Linux?

**Pull request merged**